### PR TITLE
Only show the sidebar TOC scrollbar when hovering

### PR DESCRIPTION
**Pull request by jamiemccue**
scrolling

### CODE REVIEW

**Comment by jamiemccue** [comment]
@justinbmeyer This PR is for discussion. We want to get your feedback on the functionality of the TOC sidebar scrollbars. There is a jump on hover from the width of the scroll bar but if its a concern we can fix with JS.

**Comment by jamiemccue** [comment]
@chasenlehara Based on my conversation with Justin the only thing remaining here is making the TOC area scrollable. Which is this issue: https://github.com/canjs/bit-docs-html-canjs/issues/530
He also asked to reduce the width which I did above. Is there anything else remaining?

File: static/layout.less [hunk 1]
```diff
@@ -43,6 +43,14 @@
 }
 #toc-sidebar {
   grid-area: toc-sidebar;
+  overflow: auto;
+  width: 300px;
```
Comment (chasenlehara) [inline static/layout.less]: The TOC isn’t supposed to have a set width. It’s set dynamically with [fit-content](https://developer.mozilla.org/en-US/docs/Web/CSS/fit-content):

https://github.com/canjs/bit-docs-html-canjs/blob/f5fe51a08ec578aabb60a2119ff3e6f452e38689/static/layout.less#L9

File: static/layout.less [hunk 2]
```diff
@@ -43,6 +43,14 @@
 }
 #toc-sidebar {
   grid-area: toc-sidebar;
+  overflow: auto;
```
Comment (chasenlehara) [inline static/layout.less]: This shouldn’t be necessary. The overflow is already set here: https://github.com/canjs/bit-docs-html-canjs/blob/f5fe51a08ec578aabb60a2119ff3e6f452e38689/static/layout.less#L67

**Comment by chasenlehara** [comment]
I’ve made it so space is made for the scrollbar before a user hovers over the TOC sidebar, so when they do hover over the sidebar, the layout doesn’t change:

hover

If the user changes the viewport height, the scrollbar will hide itself:

resize

If the user changes their scrollbar setting, the layout will adjust when the window is given focus:

focus

@jamiemccue, I think I have this working correctly in Chrome, Firefox, and Safari. Would you mind also testing this PR please, and also testing browsers on Windows?

**Comment by jamiemccue** [comment]
@chasenlehara Looks good! tested on Edge, FF and Chrome.

**Pull request merged**